src/main/java/org/accounting/system/util/QueryParser.java

```diff
@@ -0,0 +1,170 @@
+            JSONObject jsonObject = (JSONObject) parser.parse(json);
+
+            String type = (String) jsonObject.get("type");
+            Bson filter = null;
+            if (type.equals("filter")) {
+                filter = parseFilter(jsonObject);
+
+            } else if (type.equals("query")) {
+                filter = parseQuery(jsonObject);
+
+            }
+            if (!isAlwaysPermission) {
+                Bson accessFilter = accessFilter(filter, subOfToken, objectIds);
+                return accessFilter;
+            } else {
+                return filter;
+            }
+        }
+        return null;
+    }
+    public Bson parseFile(String json) throws ParseException, NoSuchFieldException {
+        if (json != null) {
+           JSONParser parser = new JSONParser();
+            JSONObject jsonObject = (JSONObject) parser.parse(json);
+
+            String type = (String) jsonObject.get("type");
+            Bson filter = null;
+            if (type.equals("filter")) {
+                filter = parseFilter(jsonObject);
+
+            } else if (type.equals("query")) {
+                filter = parseQuery(jsonObject);
+
+            }
+                return filter;
+
+        }
+        return null;
+    }
+
+    private Bson parseFilter(JSONObject filter) throws NoSuchFieldException {
+        String operator = (String) filter.get("operator");
+        JSONArray criteria = (JSONArray) filter.get("criteria");
+        Iterator criteriaIter = criteria.iterator();
+        ArrayList<Bson> bsons = new ArrayList<>();
+        while (criteriaIter.hasNext()) {
+            JSONObject criterio = (JSONObject) criteriaIter.next();
+            String type = (String) criterio.get("type");
+            if (type.equals("filter")) {
+                Bson bson = parseFilter(criterio);
+                bsons.add(bson);
+            } else if (type.equals("query")) {
+                Bson bson = parseQuery(criterio);
+                bsons.add(bson);
+            }
+
+
+        }
+        if (bsons.size() > 1) {
+            Bson queryBson = joinCriteria(operator, bsons);
+            return queryBson;
+        } else if (bsons.size() == 1) {
+            return bsons.get(0);
+        } else {
+            return null;
+        }
+
+    }
+
+    private Bson parseQuery(JSONObject query) {
+
+        String field = (String) query.get("field");
+        String operand = (String) query.get("operand");
+        if (operand == null) {
+            operand = "eq";
+
+        }
+        Object val = query.get("values");
+        if (val instanceof String && isDate(val)) {
+            Instant instant = Instant.parse((String) val);
+            Date time = Date.from(instant);
+            val = time;
+
+        }
+        Bson bson = convertOp(operand, field, val);
+        return bson;
+    }
+
+    private Bson joinCriteria(String operand, ArrayList<Bson> bsons) {
+
+        switch (operand) {
+
+            case "AND":
```

Comment:
Maybe it's better to use Enum instead of plain string

Reply:
fixed